cmd/integration/server.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"sync"
+)
+
+const harnessPort = "9092"
+
+// ServerConfig contains all the args and data required to launch a node
+// server instance.
+type ServerConfig struct {
+	logLevel  string
+	rpcListen string
+	exe       string
+}
+
+// DefaultConfig returns a newConfig with all default values.
+func DefaultConfig(execPath string) (*ServerConfig, error) {
+	cfg := &ServerConfig{
+		logLevel:  "debug",
+		rpcListen: "127.0.0.1:" + harnessPort,
+		exe:       execPath,
+	}
+
+	return cfg, nil
+}
+
+// genArgs generates a slice of command line arguments from ServerConfig instance.
+func (cfg *ServerConfig) genArgs() []string {
+	var args []string
+
+	args = append(args)
+
+	return args
+}
+
+// server houses the necessary state required to configure, launch,
+// and manage node server process.
+type server struct {
+	cfg *ServerConfig
+	cmd *exec.Cmd
+
+	// processExit is a channel that's closed once it's detected that the
+	// process this instance is bound to has exited.
+	processExit chan struct{}
+	// quit channel for an out source to quit process
+	quit chan struct{}
+	wg   sync.WaitGroup
+	// error channel for the server error messages
+	errChan    chan error
+	stdoutChan chan string
+}
+
+// newServer creates a new node server instance according to the passed cfg.
+func newServer(cfg *ServerConfig) (*server, error) {
+	return &server{
+		cfg:     cfg,
+		errChan: make(chan error, 5),
+	}, nil
+}
+
+// start launches a new running process of node server.
+func (s *server) start(addArgs []string) error {
+	s.quit = make(chan struct{})
+
+	args := s.cfg.genArgs()
+	// adding additional full go-spacemesh node arguments origin in
+	// yaml specification files, starting from index 1 to remove exec path
+	args = append(args, addArgs...)
+	//fmt.Println("adding arguments =", args)
+
+	s.cmd = exec.Command(s.cfg.exe, args...)
+	// Redirect stderr and stdout output to current harness buffers
+	s.cmd.Stderr = os.Stderr
+	s.cmd.Stdout = os.Stdout
+
+	// start go-spacemesh server
+	if err := s.cmd.Start(); err != nil {
+		return err
+	}
+
+	// Launch a new goroutine that bubbles up any potential fatal
+	// process errors to errChan.
+	s.processExit = make(chan struct{})
```

Comment:
who is listening to error channel?

Reply:
removed this along with the first comment